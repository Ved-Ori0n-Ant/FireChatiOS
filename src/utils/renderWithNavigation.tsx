import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {render} from '@testing-library/react-native';

const renderWithNavigation = (renderComponent: any) => {
  return render(<NavigationContainer>{renderComponent}</NavigationContainer>);
}

export default renderWithNavigation;
