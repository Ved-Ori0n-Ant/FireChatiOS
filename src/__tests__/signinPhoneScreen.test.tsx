import { NavigationContainer } from "@react-navigation/native";
import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import PhoneSigninScreen from "../screens/signedInModule/signInWithPhoneScreen";
import renderWithNavigation from "../utils/renderWithNavigation";

// function renderWithNavigation(renderComponent: any) {
//     return render(<NavigationContainer>{renderComponent}</NavigationContainer>);
// }

jest.mock('@react-native-firebase/auth', () => ({
    signInWithEmailAndPassword: jest.fn()
}));

describe('Phone signin screen should render okay!', () => {
    it('should render okay', ()=>{
        const tree = renderWithNavigation(<PhoneSigninScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should display const text', ()=>{
        const tree = renderWithNavigation(<PhoneSigninScreen />);
        const defaultText = tree.getByTestId('const-txt');

        expect(defaultText.props.children).toBe('Please enter your credentials:');
    });
    it('should have textInput', () => {
        const tree = renderWithNavigation(<PhoneSigninScreen />)
        const numInput = tree.getByTestId('phone-input');
        expect(numInput.props.placeholder).toBe('Enter your phone number');
    });
    it('should call generate OTP function', () => {
        const tree = renderWithNavigation(<PhoneSigninScreen />)
        const btn =  tree.getByTestId('otp_generator');
        const generateOTP = jest.fn();

        fireEvent.press(btn);

        expect(btn).toBeDefined();
        ()=>{
            expect(generateOTP).toBeCalledTimes(1);
        }
    });
    it('should render otp textInput', ()=>{
        const tree = renderWithNavigation(<PhoneSigninScreen />);
        const btn = tree.getByTestId('otp_generator');

        fireEvent.press(btn);
        () => {
            const otpInput = tree.getByTestId('otp-input');
            
            expect(otpInput.props.placeholder).toBe('Enter the OTP')
        }
    });
    it('should have signin btn and signup btn', ()=>{
        const tree = renderWithNavigation(<PhoneSigninScreen/>);
        const submitBtn = tree.getByTestId('sign_in_with_email');
        const signupBtn = tree.getByTestId('sign_up');
        
        expect(submitBtn).not.toBeNull();
        expect(signupBtn).not.toBeNull();
        expect(submitBtn).toBeDefined();
        expect(signupBtn).toBeDefined();
    });
})