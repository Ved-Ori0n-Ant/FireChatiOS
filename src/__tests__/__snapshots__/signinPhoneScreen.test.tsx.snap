// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phone signin screen should render okay! should render okay 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffe5b4af",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      [
        {
          "margin": 5,
          "padding": 3,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "black",
            "fontSize": 22,
            "textDecorationLine": "underline",
          },
          {
            "marginBottom": 62,
          },
        ]
      }
      testID="const-txt"
    >
      Please enter your credentials:
    </Text>
  </View>
  <View
    style={
      [
        {},
        {
          "alignSelf": "center",
          "borderBottomWidth": 1,
          "marginVertical": 2,
          "width": "85%",
        },
      ]
    }
  >
    <TextInput
      keyboardType="number-pad"
      onChangeText={[Function]}
      placeholder="Enter your phone number"
      placeholderTextColor="#0009"
      style={
        [
          {
            "borderBottomColor": "black",
            "borderBottomWidth": 1,
            "fontSize": 16,
            "fontWeight": "300",
            "padding": 10,
          },
          {
            "fontSize": 14,
            "fontWeight": "500",
            "height": 35,
          },
        ]
      }
      testID="phone-input"
      value=""
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "black",
        "borderColor": "black",
        "borderRadius": 16,
        "borderWidth": 0.3,
        "height": 42,
        "justifyContent": "center",
        "margin": 3,
        "marginTop": 23,
        "opacity": 1,
        "padding": 3,
        "width": "60%",
      }
    }
    testID="otp_generator"
  >
    <Text
      style={
        {
          "alignSelf": "center",
          "color": "white",
          "flex": 1,
          "fontSize": 16,
          "padding": 7,
          "textDecorationLine": "underline",
        }
      }
    >
      Generate OTP
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="sign_in_with_email"
  >
    <Text
      style={
        {
          "alignSelf": "center",
          "color": "black",
          "fontSize": 14,
          "marginBottom": 8,
        }
      }
    >
      Try another way of signin..
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="sign_up"
  >
    <Text
      style={
        {
          "alignSelf": "center",
          "color": "blue",
          "fontSize": 16,
        }
      }
    >
      New user? Sign up..
    </Text>
  </View>
  <View
    style={
      {
        "padding": 13,
      }
    }
  >
    <Text
      style={
        {
          "color": "red",
          "fontSize": 8,
          "fontWeight": "bold",
          "textDecorationLine": "none",
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
